src/io/file_descriptor.cr

```diff
@@ -11,9 +11,13 @@ class IO::FileDescriptor < IO
     @volatile_fd.get
   end
 
-  def initialize(fd, blocking = nil)
+  # Whether or not to close the file descriptor when this object is finalized.
+  property close_on_finalize
+
+  def initialize(fd, blocking = nil, close_on_finalize = true)
```

Comment:
I think `close_on_finalize` should be a named argument.
`blocking` as well but changing that would be a breaking change (we could deprecated it, though. But that would be a different change).
When introducing a new parameter, we should make it a named parameter right away.
Adding a type restriction is also helpful to describe the API in detail.

```suggestion
  def initialize(fd, blocking = nil, *, close_on_finalize : Bool = true)
```